datacube/index/_spatial.py

```diff
@@ -0,0 +1,107 @@
+# This file is part of the Open Data Cube, see https://opendatacube.org for more information
+#
+# Copyright (c) 2015-2024 ODC Contributors
+# SPDX-License-Identifier: Apache-2.0
+from typing import cast
+from odc.geo.geom import CRS, Geometry, box
+from datacube.model import Range, QueryDict, QueryField
+from datacube.utils.documents import JsonDict
+
+H_SPATIAL_KEYS = ("lon", "longitude", "x")
+V_SPATIAL_KEYS = ("lat", "latitude", "y")
+COORDS_SPATIAL_KEYS = H_SPATIAL_KEYS + V_SPATIAL_KEYS
+
+CRS_SPATIAL_KEYS = ("crs", "coordinate_reference_system")
+SPATIAL_KEYS = COORDS_SPATIAL_KEYS + CRS_SPATIAL_KEYS
+
+ALL_SPATIAL_KEYS = SPATIAL_KEYS + ("geopolygon",)
```

Comment:
The distinction between this and `SPATIAL_KEYS` feels like it could be potentially confusing. Given that it's only used once, is it worth defining as a constant?